### PR TITLE
Create the User and Database System

### PR DESCRIPTION
**Pull request by GCipry3**

### CODE REVIEW

File: Database/SQliteDatabase.cs [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+            DirectoryInfo directoryInfo = new DirectoryInfo(baseDirectory);
+
+            // Go up 3 directories from the baseDirectory
+            for (int i = 0; i < 3; i++)
+            {
+                directoryInfo = directoryInfo.Parent;
+                if (directoryInfo == null)
+                {
+                    throw new Exception("Invalid base directory: not enough parent directories");
+                }
```
Comment (GCipry3) [inline Database/SQliteDatabase.cs]: remove this

File: Database/SQliteDatabase.cs [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+        }
+        string GetDBPath()
+        {
+            string baseDirectory = AppDomain.CurrentDomain.BaseDirectory;
+            DirectoryInfo directoryInfo = new DirectoryInfo(baseDirectory);
+
+            // Go up 3 directories from the baseDirectory
+            for (int i = 0; i < 3; i++)
+            {
+                directoryInfo = directoryInfo.Parent;
```
Comment (GCipry3) [inline Database/SQliteDatabase.cs]: use something like directoryInfo = directoryInfo.Parent.Parent.Parent

File: Database/SQliteDatabase.cs [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+            }
+
+            string databasePath = Path.Combine(directoryInfo.FullName, "Database", "Resources", "Users.db");
+            return $"Data Source={databasePath}";
+        }
+
+        public DbConnection GetDBConnection()
+        {
+            DbConnection conn;
+            conn = new SQLiteConnection(connectionString);
```
Comment (GCipry3) [inline Database/SQliteDatabase.cs]: one line it

File: Database/SQliteDatabase.cs [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+            try
+            {
+                conn.Open();
+            }
+            catch { }
```
Comment (GCipry3) [inline Database/SQliteDatabase.cs]: If it raises an error try again to reopen it for 3 times

File: Users/IUser.cs [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Users
+{
+    public interface IUser
+    {
+        string Username { get; }
+        string Password { get; }
```
Comment (GCipry3) [inline Users/IUser.cs]: rethink this

File: CasinoLobby/Form1.cs [hunk 6]
```diff
@@ -52,6 +55,83 @@ private void DicesGameButton_Click(object sender, EventArgs e)
             dices.Show();
         }
 
+        private void InsertUserButton_Click(object sender, EventArgs e)
```
Comment (GCipry3) [inline CasinoLobby/Form1.cs]: Delete the test functions/buttons/labels for the database and users

File: Database/SQliteDatabase.cs [hunk 7]
```diff
@@ -0,0 +1,86 @@
...
+                catch (Exception)
+                {
+                }
+            }
+
+            return conn;
+        }
+
+
+
```
Comment (GCipry3) [inline Database/SQliteDatabase.cs]: remove blank lines

**Review by FCosmin27 — APPROVED** [review]

**Pull request merged**